packages/core/src/new-api/deployer.ts

```diff
@@ -0,0 +1,60 @@
+import type { IgnitionModuleResult, ModuleParameters } from "./types/module";
+import type { IgnitionModuleDefinition } from "./types/module-builder";
+
+import { Batcher } from "./internal/batcher";
+import { ExecutionEngine } from "./internal/execution-engine";
+import { ModuleConstructor } from "./internal/module-builder";
+import { Reconcilier } from "./internal/reconciliation/reconcilier";
+import { ExecutionStateMap } from "./types/execution-state";
+
+export class Deployer {
+  private _moduleConstructor: ModuleConstructor;
+  private _executionEngine: ExecutionEngine;
+
+  constructor() {
+    this._moduleConstructor = new ModuleConstructor();
+    this._executionEngine = new ExecutionEngine();
+  }
+
+  public async deploy(
+    moduleDefinition: IgnitionModuleDefinition<
+      string,
+      string,
+      IgnitionModuleResult<string>
+    >,
+    moduleParameters: ModuleParameters,
+    accounts: string[]
+  ) {
+    const module = this._moduleConstructor.construct(moduleDefinition);
```

Comment:
Why are we doing this here? My understanding was that we wanted `construct` and `deploy` to be entirely separate and distinct processes, so that you could take any output of `construct` and pass it into a `deploy`.

Reply:
Agreed, but as we are removing the `ModuleConstructor` I wouldn't block on this.

Reply:
I have been using `deployer` to capture the complete deployment process, but it is just a shim to let me test from the cli/tests.

Feel free to replace it with the `ModuleConstructor` refactor.